### PR TITLE
Searched model zoo in NAS space hub

### PR DESCRIPTION
**Pull request by ultmaster**
This PR adds `load_searched_model`, which enables a zoo of famous models obtained with NAS.

For first version, I added 18 models on 4 different spaces in total, including the state-of-the-art MobileNetV3Space.
All the models have been (re-)evaluated with a standard evaluation protocol, and the converted pretrained weights have been uploaded to NNI blob.

To make this happen, I made the following changes:

1. A lot of bug fixes in search space hub.
2. New API: `load_searched_model` and `fixed_arch`. I'm planning to add them to a common base class (e.g., `ModelSpace`) in next step.
3. Move `blob_utils` to `nni/common`, so that the download utilities can be shared between space hub and NAS benchmark.
4. Add `dim=-1` in evaluator, to eliminate a DeprecationWarning.
5. Support `min_depth = 0` in Repeat.

### How to test

CI is yet to be added. I plan to add it when I got the VMSS set up.

Manual test example as follows:

```python
model = MobileNetV3Space.load_searched_model('mobilenetv3-large-100', pretrained=True, download=True)

transform = transforms.Compose([
    transforms.Resize(256),
    transforms.CenterCrop(224),
    transforms.ToTensor(),
    transforms.Normalize(mean=[0.485, 0.456, 0.406],
                         std=[0.229, 0.224, 0.225])
])
dataset = ImageNet(directory, 'val', transform=transform)
dataloader = DataLoader(dataset, batch_size=batch_size, num_workers=num_workers)
model.eval()
with torch.no_grad():
    correct = total = 0
    pbar = tqdm.tqdm(dataloader, desc='Evaluating on ImageNet')
    for inputs, targets in pbar:
        logits = model(inputs)
        _, predict = torch.max(logits, 1)
        correct += (predict == targets).cpu().sum().item()
        total += targets.size(0)
        pbar.set_postfix({'correct': correct, 'total': total, 'acc': correct / total * 100})
print('Overall accuracy (top-1):', correct / total * 100)
```

### CODE REVIEW

File: nni/retiarii/hub/README.md
Comment (liuzhe-lz) [inline nni/retiarii/hub/README.md:10]: Wait, this file is source, not doc?

Comment (ultmaster) [reply nni/retiarii/hub/README.md:10]: Space hub is in a preview state and hidden from documentation. I think we'll announce it shortly after 1 or 2 more PRs are merged.

File: nni/common/blob_utils.py
Comment (QuanluZhang) [inline nni/common/blob_utils.py:95]: what is `upload_file` used for? who will use this function?

File: nni/retiarii/hub/README.md
Comment (QuanluZhang) [inline nni/retiarii/hub/README.md:81]: what is the meaning of this sentence? if the training is not reproduced, what is the metric in the above table?

File: nni/retiarii/hub/pytorch/mobilenetv3.py
Comment (QuanluZhang) [inline nni/retiarii/hub/pytorch/mobilenetv3.py:261]: what is meaning of `MaybeChoice`?

File: nni/common/blob_utils.py
Comment (matluster) [reply nni/common/blob_utils.py:95]: As in doctoring: `For maintainers to add...`

File: nni/retiarii/hub/README.md
Comment (matluster) [reply nni/retiarii/hub/README.md:81]: By converting the checkpoints provided by original author to our format, as in L77.

File: nni/retiarii/hub/pytorch/mobilenetv3.py
Comment (matluster) [reply nni/retiarii/hub/pytorch/mobilenetv3.py:261]: int or ValueChoice[int]

File: nni/retiarii/hub/README.md
Comment (QuanluZhang) [reply nni/retiarii/hub/README.md:81]: I see, in L77, would be better to say: The metric column is obtained by evaluating the checkpoints provided the original author and converted in NNI NAS format [these scripts]

**Review by QuanluZhang — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**
